src/Commands/CloneCommand.php

```diff
@@ -0,0 +1,191 @@
+<?php
+namespace Pantheon\Terminus\Commands;
+
+use Pantheon\Terminus\Site\SiteAwareInterface;
+use Pantheon\Terminus\Site\SiteAwareTrait;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Terminus\Models\Site;
+use Terminus\Models\Workflow;
+
+abstract class CloneCommand extends TerminusCommand implements SiteAwareInterface
+{
+    use SiteAwareTrait;
+
+    const PROGRESS_FORMAT = " <bg=blue;fg=black> %message% </>\n <info>Progress: %bar% <info>%percent:3s%%</info> \n <info>Elapsed: %elapsed:6s% \n Estimated Time Remaining: %estimated:-6s%</info>";
+    protected $operations = ['cloneFiles', 'cloneDatabase'];
+    protected $originEnv;
```

Comment:
To keep things standardized, all vars/properites should be done in underscore_case and function names in camelCase.